### PR TITLE
Error state

### PR DESCRIPTION
**Pull request by C0DK**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: https://github.com/reaviz/reaflow/issues/80

## What is the new behavior?

No visible change other than it actually works with what i called "deeply nested edges". The bug, however, is being adressed upstream, but it provides a temporary fix to the issue with deeply nested edges. Also i've changed the order, so edges are always drawn on top, rather than behind nodes. The example is also edited to include a so called "deeply nested edge"

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/Canvas.tsx [hunk 1]
```diff
@@ -339,8 +339,8 @@ const InternalCanvas: FC<CanvasProps & { ref?: Ref<CanvasRef> }> = forwardRef(
               }
             }}
           >
-            {layout?.edges?.map(renderEdge)}
             {layout?.children?.map(renderNode)}
+            {layout?.edges?.map(renderEdge)}
```
Comment (amcdnl) [inline src/Canvas.tsx:343]: Why did you move these? Just curious

Comment (C0DK) [reply src/Canvas.tsx:343]: Because SVG uses ordering for whether things are in front. You cannot set z-index. Imo it makes more sense that edges are rendered on top, not behind nodes.

Comment (amcdnl) [reply src/Canvas.tsx:343]: Yup - i knew that - didn't know if there was a specific reason - I think I ordered them like this for a reason but I can't remember so fine w/ me.

**Review by amcdnl — APPROVED** [review]
